subgraphs/dopex/src/entities/pool.ts

```diff
@@ -0,0 +1,558 @@
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  Bytes,
+  ethereum,
+} from "@graphprotocol/graph-ts";
+import {
+  LiquidityPool,
+  LiquidityPoolFee,
+  RewardToken,
+  Token,
+} from "../../generated/schema";
+import { EventType } from "./event";
+import {
+  increaseProtocolVolume,
+  getOrCreateProtocol,
+  updateProtocolTVL,
+  increaseProtocolPremium,
+  increaseProtocolSideRevenue,
+  increaseProtocolSupplySideRevenue,
+  updateProtocolOpenInterestUSD,
+  incrementProtocolEventCount,
+  updateProtocolOpenPositionCount,
+} from "./protocol";
+import { getOrCreateToken } from "./token";
+import {
+  BIGDECIMAL_ZERO,
+  INT_ZERO,
+  BIGINT_ZERO,
+  INT_ONE,
+  LiquidityPoolFeeType,
+  BIGDECIMAL_HUNDRED,
+  CHAIN_LINK,
+} from "../utils/constants";
+import { convertTokenToDecimal, multiArraySort } from "../utils/numbers";
+import { enumToPrefix } from "../utils/strings";
+import { Ssov } from "../../generated/DPXMonthlyCalls/Ssov";
+
+export function getOrCreateLiquidityPool(
+  event: ethereum.Event,
+  poolAddress: Address
+): LiquidityPool {
+  const protocol = getOrCreateProtocol();
```

Comment:
Only used within the if block, maybe we should move it in there

Reply:
Changed at 5f1c1e4a1ec9f954478dbc4eb30d355d1e8121b2 Thanks!